### ISSUE TITLE
Nested {{#if}}{{else if}}{{else}}{{/if}} inside {{#unless}} causes render error

### ISSUE THREAD

**Issue by laurenhamel** [zordius/lightncandy#314]
The Handlebars Code:
-------------

```handlebars
{{#unless foo}}
  {{#if bar}}
    // Some condition here
  {{else if baz}}
    // Some other condition here
   {{else}}
    // Some final condition here
  {{/if}}
{{else}}
  {{#if bar}}
    // Some condition here
  {{else if baz}}
    // Some other condition here
   {{else}}
    // Some final condition here
  {{/if}}
{{/unless}}
```

The Issue:
----------

The nested `if/else if/else` causes a rendering error when inside an `{{#unless}}` block. It appears that the `else if/else` portion of the nested block is being interpreted as if it were a part of the outer unless block, which then results in this error message:
```
Unexpect token {{/if}} ! Previous token {{#[unless]}} is not closed {{else}} only valid in if, unless, each, and #section context Unexpect token {{/unless}} ! Previous token {{#[if]}} is not closed
```

**Comment by zordius** [comment]
confirmed as a bug

**Comment by zordius** [comment]
Sorry, I do the confirmation again, can not find and issue. Here is the sample code:
```php
require('./vendor/autoload.php');
use LightnCandy\LightnCandy;
use LightnCandy\Runtime;

// The Template:
$template = <<<VAREND
{{#unless foo}}
  {{#if bar}}
    // 1.Some condition here
  {{else if baz}}
    // 2.Some other condition here
   {{else}}
    // 3.Some final condition here
  {{/if}}
{{else}}
  {{#if bar}}
    // 4.Some condition here
  {{else if baz}}
    // 5.Some other condition here
   {{else}}
    // 6.Some final condition here
  {{/if}}
{{/unless}}
VAREND;

// Helpers:
$helpers = array(
);

// Partials:
$partials = array(
);

$phpStr = LightnCandy::compile($template, array(
  // Used compile flags
  'flags' => LightnCandy::FLAG_HANDLEBARS | LightnCandy::FLAG_ERROR_EXCEPTION,
  'helpers' => $helpers,
  'partials' => $partials,
));

// echo "Generated PHP Code:\n$phpStr\n";
// print_r(LightnCandy::getContext());
// Input Data:
$data = array(
  'qoo' => 'moo'
);

// Save the compiled PHP code into a php file
file_put_contents('render.php', '<?php ' . $phpStr . '?>');

// Get the render function from the php file
$renderer = include('render.php');

echo "Result:\n" . $renderer($data, array('debug' => Runtime::DEBUG_ERROR_EXCEPTION));
```

The output:
```
Result:
    // 3.Some final condition here
```

**Comment by zordius** [comment]
Close this. Feel free to reopen this if you find any related issue, thanks.

**Comment by laurenhamel** [comment]
Since I've been trying to replicate Handlebars.js functionality within a server environment, I wanted to also mention the list of flags that I've been using: `FLAG_HANDLEBARSJS`, `FLAG_THIS`, `FLAG_ELSE`, `FLAG_RUNTIMEPARTIAL`, `FLAG_NAMEDARG`, `FLAG_PARENT`, `FLAG_ADVARNAME`, `FLAG_JSLENGTH`, `FLAG_SPVARS`, and `FLAG_RAWBLOCK`. In addition, I've been using `FLAG_ERROR_LOG` and/or `FLAG_ERROR_EXCEPTION` as needed for debugging. 

I'm not sure if the error I've run into is a result of the different combination of flags or something else, but just in case someone else lands here with the same issue, I figured I'd add a bit more background. My current project also makes use of multiple tiers of nested partials and a fairly lengthy list of custom helpers, and I'm essentially building out two systems simultaneously that need to mirror one another: one leverages Handlebars.js in a Node environment while the other requires a PHP-based Handlebars engine, hence LightnCandy. Somewhere down the line, I started noticing things on my end within LightnCandy deviating from expected and observed Handlebars.js behavior(s).
